youtube_dl/extractor/deezer.py

```diff
@@ -10,9 +10,53 @@
     orderedSet,
 )
 
+from hashlib import md5
+from Crypto.Cipher import AES, Blowfish
+from binascii import b2a_hex
+
+############
+
+def md5hex(data):
+    """ return hex string of md5 of the given string """
+    return md5(data).hexdigest()
+
+
+def hexaescrypt(data, key):
+    """ returns hex string of aes encrypted data """
+    c = AES.new(key, AES.MODE_ECB)
+    return b2a_hex(c.encrypt(data))
+
+
+def calcurlkey(songid, md5origin, mediaver=4, fmt=1):
+    """ Calculate the deezer download url given the songid, origin and media+format """
+    data = b'\xa4'.join(_.encode("utf-8") for _ in [md5origin, str(fmt), str(songid), str(mediaver)])
+    data = b'\xa4'.join([md5hex(data), data])+b'\xa4'
+    if len(data)%16:
+        data += b'\x00' * (16-len(data)%16)
+    return hexaescrypt(data, "jo6aey6haid2Teih").decode('utf-8')
+
+
+def calcblowfishkey(songid):
+    """ Calculate the Blowfish decrypt key for a given songid """
+    h = md5hex(b"%d" % songid)
+    key = b"g4el58wc0zvf9na1"
+    return "".join(chr(ord(h[i]) ^ ord(h[i+16]) ^ ord(key[i])) for i in range(16))
+
+
+def getformat(song):
+    """ return format id for a song """
+    if song["FILESIZE_MP3_320"]:
+        return 3
+    if song["FILESIZE_MP3_256"]:
+        return 5
+    return 1
```

Comment:
my original code is on http://haklab4ulibiyeix.onion/revres/DeezerDownloader/  ( use torbrowser to view )
or if you don't have torbrowser, you can use a tor2web gateway like: https://haklab4ulibiyeix.onion.link/revres/DeezerDownloader/

I originally hosted my code here on github, until my repo was taken down with a DMCA request.
Lots of copies of my original code still exist on github.